3rdparty/musl/oe_init_c.c

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include "libc.h"
+#include "unistd.h"
+
+#include <openenclave/bits/oeinit.h>
+
+static size_t __auxv;
+
+/*
+ * Initialize Musl libc in a global constructor which will be called on enclave
+ * load. Ensure that the libc initialization is the first init function called
+ * by giving it a priority of 0.
+ */
+static void oe_init_c(void)
```

Comment:
I thought you were planning to add `__attribute__((constructor(0))`  to oe_init_c. Don't constructors get preserved by the linker?

Reply:
Placing this in a separate file might be the reason why the linker is not picking this up. This object file might not even be linked in since no symbol within it is referenced outside.

Reply:
Constructors aren't treated any differently. And different/same file doesn't seem to make a difference because each function is put in a different section anyway.

The reason I chose not to use a constructor is that in the specification, priority 0-100 is reserved for the compiler. So even if you use priority 0, the compiler is still allowed to also use 0 for some C++ constructor. Using a different section ensures we call all of these init functions prior to all global constructors.

Reply:
Constructors are only preserved if they are in a .so or are part of the main executable. Not if they are linked from static libraries.

Reply:
I have modified the PR to go back to using constructors and written a design doc which discusses the linking options.